pkg/runner/instance/conn.go

```diff
@@ -29,64 +31,70 @@ var (
 	splitByte = []byte("littledrizzle")
 )
 
-// functionRequest will be put into the producer and send it to request pipe
-type functionRequest struct {
-	Id         string `json:"id"`
-	Parameters string `json:"parameters"`
+// FunctionRequest will be put into the producer and send it to request pipe
+type FunctionRequest struct {
+	Id         string                 `json:"id"`
+	Parameters map[string]interface{} `json:"parameters"`
 }
 
-func NewFunctionRequest(id string, parameters map[string]interface{}) *functionRequest {
-	parameterStr, err := json.Marshal(parameters)
+func NewFunctionRequest(id string, parameters map[string]interface{}) *FunctionRequest {
+	transfer, err := common.CopyMap(parameters)
 	if err != nil {
 		zap.S().Errorw("functionRequest marshal error", "err", err)
 		return nil
 	}
-	return &functionRequest{
+	return &FunctionRequest{
 		Id:         id,
-		Parameters: string(parameterStr),
+		Parameters: transfer,
 	}
 }
 
-// functionResponse will be put into channel to be consume by instance.
-type functionResponse struct {
+// FunctionResponse will be put into channel to be consume by instance.
+type FunctionResponse struct {
 	Id     string
 	Result map[string]interface{}
 }
 
-type producer struct {
+type Producer struct {
 	f              *os.File
-	requestChannel chan functionRequest
+	demo           interface{}
```

Comment:
What's this? data?

Reply:
the demo just given the `Type`.
This change was made to be more general